File: Java/257. Binary Tree Paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> res = new ArrayList();
        String s = "";
        helper(root,s,res);
        return res;
    }
    void helper(TreeNode root,String s,List<String> res){
        if(root.left==null && root.right==null){
            s=s+root.val;
            res.add(s);
        }
        s=s+root.val+"->";
        if(root.left!=null){
            helper(root.left,s,res);
        }
        if(root.right!=null){
            helper(root.right,s,res);
        }
    }
}

File: C++/257. Binary Tree Paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> res;
        string s = "";
        helper(root,s,res);
        return res;
    }
    void helper(TreeNode* root,string s,vector<string> &res){
        if(root->left==NULL && root->right==NULL){
            s=s+to_string(root->val);
            res.push_back(s);
        }
        s=s+to_string(root->val)+"->";
        if(root->left!=NULL){
            helper(root->left,s,res);
        }
        if(root->right!=NULL){
            helper(root->right,s,res);
        }
    }
};
